utils/prof_utils.py

```diff
@@ -0,0 +1,523 @@
+#!/usr/bin/env python3
+
+import numpy as np
+import re
+import sys
+import sys
+import logging
+import argparse
+from scipy.interpolate import UnivariateSpline
+import matplotlib
+matplotlib.use('Agg')
+import matplotlib.pyplot as plt
+from astropy.time import Time
+
+from stickel import Stickel
+
+logger = logging.getLogger(__name__)
+
+def get_from_bestprof(file_loc):
+    """
+    Get info from a bestprof file
+
+    Parameters:
+    -----------
+    file_loc: string
+        The path to the bestprof file
+    
+    Returns:
+    --------
+    [obsid, pulsar, dm, period, period_uncer, obsstart, obslength, profile, bin_num]: list
+        obsid: int
+            The observation ID
+        pulsar: string
+            The name of the pulsar
+        dm: float
+            The dispersion measure of the pulsar
+        period: float
+            The period of the pulsar
+        period_uncer: float
+            The uncertainty in the period measurement
+        obsstart: int
+            The beginning time of the observation
+        obslength: float
+            The length of the observation in seconds
+        profile: list
+            A list of floats containing the profile data
+        bin_num: int
+            The number of bins in the profile
+    """
+
+    with open(file_loc,"r") as bestprof:
+        lines = bestprof.readlines()
+        # Find the obsid by finding a 10 digit int in the file name
+        obsid = re.findall(r'(\d{10})', lines[0])[0]
+        try:
+            obsid = int(obsid)
+        except:
+            obsid = None
+
+        pulsar = str(lines[1].split("_")[-1][:-1])
+        if not (pulsar.startswith('J') or pulsar.startswith('B')):
+            pulsar = 'J{0}'.format(pulsar)
+
+        dm = lines[14][22:-1]
+
+        period = lines[15][22:-1]
+        period, period_uncer = period.split('  +/- ')
+
+        mjdstart = Time(float(lines[3][22:-1]), format='mjd', scale='utc')
+        # Convert to gps time
+        obsstart = int(mjdstart.gps)
+
+        # Get obs length in seconds by multipling samples by time per sample
+        obslength = float(lines[6][22:-1])*float(lines[5][22:-1])
+
+        # Get the pulse profile
+        orig_profile = []
+        for l in lines[27:]:
+            orig_profile.append(float(l.split()[-1]))
+        bin_num = len(orig_profile)
+        profile = np.zeros(bin_num)
+
+        # Remove min
+        min_prof = min(orig_profile)
+        for p in range(len(orig_profile)):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider using enumerate instead of iterating with range and len](https://app.codacy.com/app/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4645369)